### ISSUE TITLE
DARE and TIES not working for Mixtral.

### ISSUE THREAD

**Issue by PhilipMay** [arcee-ai/mergekit#210]
Hi,

I am using DARE and TIES for Mixtral models.
It does not work. The models provide no output.

Anybody else ever tried to merge multiple Mixtral models?
Is this a bug or on my side?

**Comment by Aratako** [comment]
I think this is because the current MixtralTensorNames class is lacking some tensors.
Now the implementation is like below in [architecture.py](https://github.com/arcee-ai/mergekit/blob/main/mergekit/architecture.py)
```python
class MixtralTensorNames(ArchitectureInfo, BaseModel):
    ARCHITECTURE_NAME: ClassVar[str] = "MixtralForCausalLM"
    num_local_experts: int

    @classmethod
    def from_config(cls, config: PretrainedConfig):
        return MixtralTensorNames(num_local_experts=config.num_local_experts)

    def pre_weights(self, config: PretrainedConfig) -> List[WeightInfo]:
        return MISTRAL_INFO.pre_weights(config)

    def post_weights(self, config: PretrainedConfig) -> List[WeightInfo]:
        return MISTRAL_INFO.post_weights(config)

    def num_layers_config_key(self) -> str:
        return MISTRAL_INFO.num_layers_config_key()

    def layer_weights(
        self, index: int, config: PretrainedConfig
    ) -> Optional[List[WeightInfo]]:
        num_experts = self.num_local_experts
        prefix = f"model.layers.{index}"
        tensor_names = []
        for expert_idx in range(num_experts):
            for param in ("w1", "w2", "w3"):
                tensor_names.append(
                    prefix + f".block_sparse_moe.experts.{expert_idx}.{param}.weight"
                )
        tensor_names.append(prefix + ".block_sparse_moe.gate.weight")
        res = []
        for name in tensor_names:
            res.append(WeightInfo(name=name))
        return res

    def sliceable(self) -> bool:
        return True

    def has_defined_spaces(self) -> bool:
        return False
```

It looks like this class is lacking these 6 weights.

```
"model.layers.{index}.input_layernorm.weight"
"model.layers.{index}.post_attention_layernorm.weight"
"model.layers.{index}.self_attn.k_proj.weight"
"model.layers.{index}.self_attn.o_proj.weight"
"model.layers.{index}.self_attn.q_proj.weight"
"model.layers.{index}.self_attn.v_proj.weight"
```

You can add these manually like below code:
```python
    def layer_weights(
        self, index: int, config: PretrainedConfig
    ) -> Optional[List[WeightInfo]]:
        num_experts = self.num_local_experts
        prefix = f"model.layers.{index}"
        tensor_names = []
        for expert_idx in range(num_experts):
            for param in ("w1", "w2", "w3"):
                tensor_names.append(
                    prefix + f".block_sparse_moe.experts.{expert_idx}.{param}.weight"
                )
        tensor_names.append(prefix + ".block_sparse_moe.gate.weight")
        tensor_names.append(prefix + ".input_layernorm.weight")
        tensor_names.append(prefix + ".post_attention_layernorm.weight")
        tensor_names.append(prefix + ".self_attn.k_proj.weight")
        tensor_names.append(prefix + ".self_attn.o_proj.weight")
        tensor_names.append(prefix + ".self_attn.q_proj.weight")
        tensor_names.append(prefix + ".self_attn.v_proj.weight")
        res = []
        for name in tensor_names:
            res.append(WeightInfo(name=name))
        return res
```

I have only tried passthrough merging, but at least with that, it seems to be working correctly.

**Comment by cg123** [comment]
Well that's embarrassing. I must have messed this up in a merge a while back.

This should be resolved by #212.
